### DISCUSSION TITLE
How to add local node logs to docker-composer?

### DISCUSSION THREAD

**Discussion by pomazanbohdan** [netdata/netdata#16455] [Q&A] [answered]
How to add local node logs to docker-composer?

**Comment by ilyam8** [comment]
Hey, @pomazanbohdan. What local logs are you referring to? Can you please elaborate?

**Reply by pomazanbohdan** [reply]
Systemd-journal from last release

**Reply by ilyam8** [reply, marked as answer]
Ah, I see. Unfortunately, the Netdata docker image does not have the systemd-journal plugin due to limitations of Alpine (base image). We plan to switch to Debian-based images (I hope) soon ([PR](https://github.com/netdata/netdata/pull/15823)).

**Comment by ilyam8** [comment]
@pomazanbohdan #15823 has been merged. Tomorrow's nightly docker image will be based on Debian. Systemd-journal plugin will be working. A note: an additional mount `/var/log:/host/var/log:ro` is required (docs update #16496).

**Comment by pomazanbohdan** [comment]
I update docker-compose file, pull new image, but 

```
Required filters are needed
Function "systemd-journal" needs: source.
Check them on the right sidebar!
```

P.S. ./netdataconfig:/etc/netdata

**Reply by ilyam8** [reply]
What sources do you have on the right sidebar? Have you mounted /var/log to /host/var/log?

**Comment by pomazanbohdan** [comment]
![image](https://github.com/netdata/netdata/assets/696713/6afe1974-d87e-4ba5-9ed1-aa481aa3e67f)

**Comment by pomazanbohdan** [comment]
![image](https://github.com/netdata/netdata/assets/696713/153965f5-5e8e-411a-81b2-d09750d01366)

**Reply by ilyam8** [reply]
Sources:

<details>
<summary>Screenshot</summary>

<img width="1642" alt="Screenshot 2023-11-30 at 14 29 03" src="https://github.com/netdata/netdata/assets/22274335/16514003-4cb2-4866-9e1e-5c49f9547ba7">

</details>

I think the issue is that journald on your system doesn't use persistent storage for logs. See the [Storage config option](https://www.freedesktop.org/software/systemd/man/latest/journald.conf.html#Storage=).

TL;DR

> Controls where to store journal data.
> One of "**volatile**", "**persistent**", "**auto**" and "**none**".
> If "**persistent**", data will be stored preferably on disk, i.e. below the /var/log/journal hierarchy (which is created if needed), with a fallback to /run/log/journal (which is created if needed), during early boot and if the disk is not writable. 
> "**auto**" behaves like "persistent" if the /var/log/journal directory exists, and "volatile" otherwise (the existence of the directory controls the storage mode).


So in your case, the logs are in `/run/log/journal` not in `/var/log/journal`. You have two options:
 - if don't want to use persistent storage, mount `/run/log` to `/host/run/log`.
 - switch to persistent by setting Storage to persistent (or creating `/var/log/journal` + setting ownership to root:systemd-journal) and mount `/var/log/:/host/var/log`

**Comment by pomazanbohdan** [comment]
> * if don't want to use persistent storage, mount `/run/log` to `/host/run/log`.

Thank you, this helped me

**Reply by ilyam8** [reply]
Also, in order to make working [go.d/systemdunits](https://github.com/netdata/go.d.plugin/tree/master/modules/systemdunits#readme) collector and Systemd-list-units functions: add `/run/dbus:/run/dbus:ro` mount.
